desc/objectives/_geometry.py

```diff
@@ -750,24 +791,324 @@ def compute(self, equil_params, surface_params=None, constants=None):
+                    2,
+                    quads,
+                )
+                quads = jnp.where(
+                    jnp.logical_and(Rbool, jnp.logical_not(Zbool)), 3, quads
+                )
+                deltas = quads[1:, :] - quads[0:-1, :]
+                deltas = jnp.where(deltas == 3, -1, deltas)
+                deltas = jnp.where(deltas == -3, 1, deltas)
+                # then flip sign if the R intercept is > Rtest and the
+                # quadrant flipped over a diagonal
+                b = (Z[1:] / R[1:] - Z[0:-1] / R[0:-1]) / (Z[1:] - Z[0:-1])
+                Rint = Rtest[:, None] - b * (R[1:] - R[0:-1]) / (Z[1:] - Z[0:-1])
+                deltas = jnp.where(
+                    jnp.logical_and(jnp.abs(deltas) == 2, Rint > Rtest),
+                    -deltas,
+                    deltas,
+                )
+                pt_sign = jnp.sum(deltas, axis=0)
+                # positive distance if the plasma pt is inside the surface, else
+                # negative distance is assigned
+                # pt_sign = 0 : Means SURFACE is OUTSIDE of the PLASMA,
+                #               assign positive distance
+                # pt_sign = +/-4: Means SURFACE is INSIDE PLASMA, so
+                #                 assign negative distance
+                pt_sign = jnp.where(jnp.isclose(pt_sign, 0), 1, -1)
+                return pt_sign
+
+            # loop over zeta planes
+            for plasma_zeta_idx, surface_zeta_idx in zip(
+                constants["plasma_zeta_indices"], constants["surface_zeta_indices"]
+            ):
+                plasma_pts_at_zeta_plane = plasma_coords_rpz[plasma_zeta_idx, :]
+                surface_pts_at_zeta_plane = surface_coords_rpz[surface_zeta_idx, :]
+                plasma_pts_at_zeta_plane = jnp.vstack(
+                    (plasma_pts_at_zeta_plane, plasma_pts_at_zeta_plane[0, :])
+                )
+
+                surface_pts_at_zeta_plane
+                pt_sign = _find_angle_vec(
+                    plasma_pts_at_zeta_plane[:, 0],
+                    plasma_pts_at_zeta_plane[:, 2],
+                    surface_pts_at_zeta_plane[:, 0],
+                    surface_pts_at_zeta_plane[:, 2],
+                )
+
+                # need to assign to correct index of the points on the surface
+                point_signs = point_signs.at[surface_zeta_idx].set(pt_sign)
+            # at end here, point_signs is either +/- 1  with
+            # positive meaning the surface pt
+            # is outside the plasma and -1 if the surface pt is
+            # inside the plasma
 
         if self._use_softmin:  # do softmin
-            return jnp.apply_along_axis(softmin, 0, d, self._alpha)
+            return jnp.apply_along_axis(softmin, 0, d, self._alpha) * point_signs
         else:  # do hardmin
-            return d.min(axis=0)
+            return d.min(axis=0) * point_signs
```

Comment:
Does hard min makes this whole objective a scalar? In that case we also have to take the sign from that minimum point, because I think `point_signs` here is still a vector. 

What axis is soft min taking the minimum along? In other words, what is the dimension of the output with soft min? 

This objective always uses `dim_f = surface_grid.num_nodes` so I think there are some inconsistencies here.

Reply:
No, this PR does not change any dimensions. The `d` object here in both cases is a 2D array of dimensions [N_plasma, N_surf] (or vice versa, though I think it is this one), and it contains the matrix of distances between every plasma point and every surface point. The reduction operation to a "min" distance can either be, for each surface point, the min of the dist btwn that point and every point on the plasma, or the softmin function of the dist btwn that point and every point on the plasma.

both will return `surface_grid.num_nodes` sized array, so there are no inconsistencies.